File: app/src/main/java/com/first/myapplication/mht/SplashActivity.java
package com.first.myapplication.mht;

import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.os.Bundle;
import android.os.Handler;
import android.util.Log;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.constraintlayout.widget.ConstraintLayout;

import com.firebase.client.Firebase;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class SplashActivity extends AppCompatActivity {

    FirebaseAuth mAuth;
    Firebase firebase;

    TextView tvNoInternet;
    Button buttonRefresh;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        Firebase.setAndroidContext(this);
        mAuth = FirebaseAuth.getInstance();

        tvNoInternet = findViewById(R.id.tvNoInternet);
        buttonRefresh = findViewById(R.id.btnRefresh);

        if(isConnected()){
            tvNoInternet.setVisibility(View.GONE);
            buttonRefresh.setVisibility(View.GONE);
            Handler handler = new Handler();
            handler.postDelayed(new Runnable() {
                @Override
                public void run() {
                    FirebaseUser user = mAuth.getCurrentUser();
                    updateUI(user);
                }
            }, 1000);
        }
        else{
            tvNoInternet.setVisibility(View.VISIBLE);
            buttonRefresh.setVisibility(View.VISIBLE);
        }

        buttonRefresh.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                recreate();
            }
        });
    }

    public void updateUI(FirebaseUser user) {
        if (user != null) {
            Intent i = new Intent(SplashActivity.this, themeActivity.class);
            startActivity(i);
        } else {
            Intent i = new Intent(SplashActivity.this, MainActivity.class);
            startActivity(i);
            finish();
        }
    }

    public boolean isConnected() {
        boolean connected = false;
        try {
            ConnectivityManager cm = (ConnectivityManager)getApplicationContext().getSystemService(Context.CONNECTIVITY_SERVICE);
            NetworkInfo nInfo = cm.getActiveNetworkInfo();
            connected = nInfo != null && nInfo.isAvailable() && nInfo.isConnected();
            return connected;
        } catch (Exception e) {
            Log.e("Connectivity Exception", e.getMessage());
        }
        return connected;
    }
}

File: app/src/main/java/com/first/myapplication/mht/SplashActivity.kt
package com.first.myapplication.mht

import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager
import android.os.Bundle
import android.os.Handler
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.firebase.client.Firebase
import com.first.myapplication.mht.utils.isConnectionAvailable
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser
import kotlinx.android.synthetic.main.activity_splash.*

class SplashActivity : AppCompatActivity() {

    var mAuth: FirebaseAuth? = null
    var firebase: Firebase? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        Firebase.setAndroidContext(this)
        mAuth = FirebaseAuth.getInstance()

        if (this.isConnectionAvailable()) {
            tvNoInternet.visibility = View.GONE
            btnRefresh.visibility = View.GONE
            val handler = Handler()
            handler.postDelayed({
                val user = mAuth!!.currentUser
                updateUI(user)
            }, 700)
        } else {
            tvNoInternet.visibility = View.VISIBLE
            btnRefresh.visibility = View.VISIBLE
        }
        btnRefresh.setOnClickListener(View.OnClickListener { recreate() })
    }

    private fun updateUI(user: FirebaseUser?) {
        if (user != null) {
            val i = Intent(this@SplashActivity, PsychoThemeActivity::class.java)
            startActivity(i)
        } else {
            val i = Intent(this@SplashActivity, GettingStartedActivity::class.java)
            startActivity(i)
            finish()
        }
    }
}